packages/devtools_app/lib/src/debugger/call_stack.dart

```diff
@@ -33,24 +33,37 @@ class _CallStackState extends State<CallStack> {
 
   @override
   Widget build(BuildContext context) {
-    return ValueListenableBuilder<List<StackFrameAndSourcePosition>>(
-      valueListenable: controller.stackFramesWithLocation,
-      builder: (context, stackFrames, _) {
-        return ValueListenableBuilder<StackFrameAndSourcePosition>(
-          valueListenable: controller.selectedStackFrame,
-          builder: (context, selectedFrame, _) {
-            return ListView.builder(
-              itemCount: stackFrames.length,
-              itemExtent: defaultListItemHeight,
-              itemBuilder: (_, index) {
-                final frame = stackFrames[index];
-                return _buildStackFrame(frame, frame == selectedFrame);
+    return Column(children: [
+      ValueListenableBuilder<List<StackFrameAndSourcePosition>>(
+          valueListenable: controller.stackFramesWithLocation,
+          builder: (context, stackFrames, _) {
+            return Expanded(
+                child: ValueListenableBuilder<StackFrameAndSourcePosition>(
+              valueListenable: controller.selectedStackFrame,
+              builder: (context, selectedFrame, _) {
+                return ListView.builder(
+                  itemCount: stackFrames.length,
+                  itemExtent: defaultListItemHeight,
+                  itemBuilder: (_, index) {
+                    final frame = stackFrames[index];
+                    return _buildStackFrame(frame, frame == selectedFrame);
+                  },
+                );
               },
-            );
-          },
-        );
-      },
-    );
+            ));
+          }),
+      ValueListenableBuilder<bool>(
```

Comment:
it would be nice to avoid having a SHOW ALL button and instead load the stack frames when they are needed. All you have to do to make that happen is trigger the load when a frame that isn't available yet is requested in the itemBuilder callback.

Reply:
I think that's a fine improvement but should be left as a follow up PR. Note that it can take several seconds to compute the full stack so we'll want to give some indication to the user when it is truncated, and when we are loading more frames.

Reply:
Ah, this was the feedback I thought of when I saw the screen shot above. This would definitely be a useful improvement.

In terms of displaying progress, perhaps something like a circular progress indicator in the top right of the call stack (in the area where we display actions for other views like the breakpoints view).